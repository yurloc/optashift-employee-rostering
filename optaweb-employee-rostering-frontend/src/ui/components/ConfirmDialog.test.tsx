/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import { ConfirmDialogProps, ConfirmDialog } from './ConfirmDialog';

describe('ConfirmDialog component', () => {
  it('should render correctly when closed', () => {
    const confirmDialogComponent = shallow(<ConfirmDialog {...confirmDialogProps} isOpen={false}>Body</ConfirmDialog>);
    expect(toJson(confirmDialogComponent)).toMatchSnapshot();
  });

  it('should render correctly when opened', () => {
    const confirmDialogComponent = shallow(<ConfirmDialog {...confirmDialogProps}>Body</ConfirmDialog>);
    expect(toJson(confirmDialogComponent)).toMatchSnapshot();
  });

  it('should call onClose when close or the cross is clicked', () => {
    const confirmDialogComponent = shallow(<ConfirmDialog {...confirmDialogProps}>Body</ConfirmDialog>);
    confirmDialogComponent.find('Modal[title="Confirm Dialog Title"]').simulate('close');
    expect(confirmDialogProps.onClose).toBeCalled();
    expect(confirmDialogProps.onConfirm).not.toBeCalled();

    jest.resetAllMocks();
    shallow((confirmDialogComponent.find('Modal[title="Confirm Dialog Title"]')
      .prop('actions') as any[])[0]).simulate('click');
    expect(confirmDialogProps.onClose).toBeCalled();
    expect(confirmDialogProps.onConfirm).not.toBeCalled();
  });

  it('should call both onClose and onConfirm when the confirm button is clicked', () => {
    const confirmDialogComponent = shallow(<ConfirmDialog {...confirmDialogProps}>Body</ConfirmDialog>);
    shallow((confirmDialogComponent.find('Modal[title="Confirm Dialog Title"]')
      .prop('actions') as any[])[1]).simulate('click');
    expect(confirmDialogProps.onClose).toBeCalled();
    expect(confirmDialogProps.onConfirm).toBeCalled();
  });
});

const confirmDialogProps: ConfirmDialogProps = {
  title: 'Confirm Dialog Title',
  isOpen: true,
  onConfirm: jest.fn(),
  onClose: jest.fn(),
};
