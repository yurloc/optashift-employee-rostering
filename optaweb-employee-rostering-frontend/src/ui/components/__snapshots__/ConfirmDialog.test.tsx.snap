// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDialog component should render correctly when closed 1`] = `
<Modal
  actions={
    Array [
      <Button
        aria-label="Close Modal"
        onClick={[MockFunction]}
        variant="tertiary"
      >
        Trans(i18nKey=close)
      </Button>,
      <Button
        aria-label="Trans(i18nKey=confirm)"
        data-cy="confirm"
        onClick={[Function]}
      >
        Trans(i18nKey=confirm)
      </Button>,
    ]
  }
  ariaDescribedById=""
  className=""
  hideTitle={false}
  isLarge={false}
  isOpen={false}
  isSmall={true}
  onClose={[MockFunction]}
  showClose={true}
  title="Confirm Dialog Title"
>
  Body
</Modal>
`;

exports[`ConfirmDialog component should render correctly when opened 1`] = `
<Modal
  actions={
    Array [
      <Button
        aria-label="Close Modal"
        onClick={[MockFunction]}
        variant="tertiary"
      >
        Trans(i18nKey=close)
      </Button>,
      <Button
        aria-label="Trans(i18nKey=confirm)"
        data-cy="confirm"
        onClick={[Function]}
      >
        Trans(i18nKey=confirm)
      </Button>,
    ]
  }
  ariaDescribedById=""
  className=""
  hideTitle={false}
  isLarge={false}
  isOpen={true}
  isSmall={true}
  onClose={[MockFunction]}
  showClose={true}
  title="Confirm Dialog Title"
>
  Body
</Modal>
`;
